libvast/vast/detail/fill_status_map.hpp

```diff
@@ -0,0 +1,24 @@
+/******************************************************************************
+ *                    _   _____   __________                                  *
+ *                   | | / / _ | / __/_  __/     Visibility                   *
+ *                   | |/ / __ |_\ \  / /          Across                     *
+ *                   |___/_/ |_/___/ /_/       Space and Time                 *
+ *                                                                            *
+ * This file is part of VAST. It is subject to the license terms in the       *
+ * LICENSE file found in the top-level directory of this distribution and at  *
+ * http://vast.io/license. No part of VAST, including this file, may be       *
+ * copied, modified, propagated, or distributed except according to the terms *
+ * contained in the LICENSE file.                                             *
+ ******************************************************************************/
+
+#pragma once
+
+#include <caf/fwd.hpp>
+
+namespace vast::detail {
+
+/// Fills `xs` with state from the stream manager `mgr`.
+void fill_status_map(caf::dictionary<caf::config_value>& xs,
```

Comment:
Output reference parameters are quite ugly, is there a change to add something like [map::merge](https://stackoverflow.com/questions/53205898/how-do-i-merge-two-maps-in-stl-and-apply-a-function-for-conflicts) to caf and expose it for `caf::dictionary`?
Then this function could just return a dictionary and the calling code can combine it with additional data.

Reply:
Note: We already have something very close to that in vast at `set_operations.hpp`.

Reply:
Unfortunately, `map::merge` is a C++17 feature. Since CAF is still at C++11, we have to live without it for now. I'd love to get rid of the output parameter as well, but not at the cost of doubling allocations.